services/cases-api/lambdas/createCase.js

```diff
@@ -22,43 +22,61 @@ export async function main(event) {
   const [validationError, validatedEventBody] = await to(
     validateCreateCaseRequestBody(event.body, caseValidationSchema)
   );
-
   if (validationError) {
     return response.failure(validationError);
   }
 
   // TODO: check if the passed formId exsists in the form dynamo table.
 
+  const { formId, provider, ...rest } = validatedEventBody;
   const { personalNumber } = decodedToken;
   const id = uuid.v4();
+  const PK = `USER#${personalNumber}`;
+  const SK = `USER#${personalNumber}#CASE#${id}`;
+  const timestamp = Date.now();
+
+  const Item = {
+    PK,
+    SK,
+    id,
+    createdAt: timestamp,
+    updatedAt: timestamp,
+    status: CASE_STATUS_ONGOING,
+    details: {},
+    answers: [],
+    formId,
+    provider,
+    ...rest,
+  };
 
-  const caseItem = {
+  const putItemParams = {
     TableName: config.cases.tableName,
-    Item: {
-      PK: `USER#${personalNumber}`,
-      SK: `USER#${personalNumber}#CASE#${id}`,
-      id,
-      createdAt: Date.now(),
-      updatedAt: Date.now(),
-      status: CASE_STATUS_ONGOING,
-      details: {},
-      answers: [],
-      ...validatedEventBody,
-    },
+    Item,
   };
 
-  // TODO: putItem does not return data response from dynamodb value due to promise issues.
-  const [putItemError] = await to(putItem(caseItem));
+  const [putItemError] = await to(putItem(putItemParams));
   if (putItemError) {
     return response.failure(putItemError);
   }
 
-  // TODO: response from dynamodb should be used here instead since that's the source of truth after creation of a case.
-  const { PK, SK, ...attributes } = caseItem.Item;
+  // Since we cannot use the ReturnValues attribute on the DynamoDB putItem operation to return the created item,
+  // we need to do a second request in order to retrive the item/case after successfull creation.
+  // This can be found in the AWS docs https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html#DDB-PutItem-request-ReturnValues
+  const [
+    getItemError,
+    {
+      Item: { PK: pk, SK: sk, ...caseItem },
+    },
+  ] = await getItem(config.cases.tableName, PK, SK);
+  if (getItemError) {
+    return response.failure(getItemError);
+  }
 
   return response.success(201, {
     type: 'createCases',
-    attributes,
+    attributes: {
+      ...caseItem,
+    },
```

Comment:
What do I get from caseItem?

Reply:
Updated the definition of Item and response for better readability